modules/blocks.php

```diff
@@ -60,6 +66,35 @@ function jetpack_tiled_gallery_load_block_assets( $attr, $content ) {
 	}
 }
 
+/**
+ * Giphy block registration/dependency declaration.
+ *
+ * @param array $attr - Array containing the map block attributes.
+ *
+ * @return string
+ */
+function jetpack_giphy_block_load_assets( $attr ) {
+	$align       = isset( $attr['align'] ) ? $attr['align'] : 'center';
+	$style       = 'padding-top:' . $attr['paddingTop'];
+	$giphy_url   = isset( $attr['giphyUrl'] ) ? $attr['giphyUrl'] : '//giphy.com/embed/ZgTR3UQ9XAWDvqy9jv';
+	$search_text = isset( $attr['searchText'] ) ? $attr['searchText'] : '';
+	$caption     = isset( $attr['caption'] ) ? $attr['caption'] : null;
+	ob_start();
+	?>
+	<div class="wp-block-jetpack-giphy align<?php echo( esc_attr( $align ) ); ?>">
+		<figure style=<?php echo( esc_attr( $style ) ); ?>>
+			<iframe src='<?php echo( esc_attr( $giphy_url ) ); ?>' title='<?php echo( esc_attr( $search_text ) ); ?>'></iframe>
+		</figure>
+		<?php if ( $caption ) : ?>
+			<figcaption class="caption"><?php echo( esc_html( $caption ) ); ?></figcaption>
```

Comment:
🤔 This is a concern. We can't `esc_html` here because it's RichText in the editor. We _expect_ HTML.

Should we use [`wp_kses`](https://developer.wordpress.org/reference/functions/wp_kses/)?

Also, `figcaption` must be first or last child of `figure` [according to MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption). We'll need to rework the layout a bit.

Are we planning to keep captions, or can we forget this question altogether?

Reply:
Tracking the captions or no question: https://github.com/Automattic/wp-calypso/issues/30345

Reply:
We've decided to keep captions.

I used [`wp_kses_post`](https://developer.wordpress.org/reference/functions/wp_kses_post/) to escape this in c412741.